api/src/legacy_soap_api/applicants/services/get_opportunity_list_response.py

```diff
@@ -38,6 +50,47 @@ def get_competitions_from_opportunity_list_request(
     )
 
 
+def get_tinstructions_map(db_session: db.Session, competitions: list) -> dict:
+    """Get instructions extension for competitions
+
+    This method returns a map of extension string from the staging.tinstructions table
+    that is needed to build the instructions_url property of the competition for the
+    GetOpportunityListResponse.
+
+    This will return a map of Competition.legacy_competition_id: staging.tinstructions.extension.
+    """
+    legacy_competition_ids = [
+        c.legacy_competition_id for c in competitions if c.legacy_competition_id is not None
+    ]
+    tinstructions = (
+        db_session.execute(
+            select(Tinstructions.extension, Tinstructions.comp_id).where(
+                Tinstructions.comp_id.in_(legacy_competition_ids)
+            )
+        )
+        .mappings()
+        .all()
+    )
+    return {tinstruction.comp_id: tinstruction.extension for tinstruction in tinstructions}
+
+
+def get_tinstructions_urls(competition: Competition, tinstructions_map: dict) -> TinstructionsURL:
+    urls = TinstructionsURL()
+    if not competition.legacy_package_id:
+        return urls
+    base_url = file_util.join(get_soap_config().grants_gov_uri, "apply", "opportunities")
+    urls.schema_url = file_util.join(
+        base_url, "schemas", "applicant", f"{competition.legacy_package_id}.xsd"
+    )
+    if extension := tinstructions_map.get(competition.legacy_competition_id):
+        urls.instructions_url = file_util.join(
+            base_url,
+            "instructions",
+            f"{competition.legacy_package_id}-instructions{ensure_dot_prefix(extension)}",
+        )
+    return urls
```

Comment:
Just to go alongside renaming the class, could you make these all `url` instead of `urls`?